python/cudf/cudf/core/column/column.py

```diff
@@ -2140,15 +2151,9 @@ def as_column(
                 arbitrary,
                 from_pandas=True,
             )
-            if isinstance(pyarrow_array.type, pa.Decimal128Type):
-                pyarrow_type = cudf.Decimal128Dtype.from_arrow(
-                    pyarrow_array.type
-                )
-            else:
-                pyarrow_type = arbitrary.dtype
```

Comment:
Do we not still need this logic if the provided dtype was None?

Reply:
I don't think so. if `dtype=None`, the dtype inference will be done by pyarrow and will be picked up by `ColumnBase`.  Plus we don't want to override the user's passed `dtype` if it wasn't `None`